src/app/tests/suites/certification/Test_TC_S_2_3.yaml

```diff
@@ -20,99 +20,581 @@ PICS:
 
 config:
     nodeId: 0x12344321
-    cluster: "Basic Information"
-    endpoint: 0
+    cluster: "Scenes"
+    endpoint: 1
 
 tests:
+    - label: "Wait for the commissioned device to be retrieved"
+      cluster: "DelayCommands"
+      command: "WaitForCommissionee"
+      arguments:
+          values:
+              - name: "nodeId"
+                value: nodeId
+
+    - label: "TH reads from the DUT the (0x0006) SceneTableSize attribute"
+      PICS: S.S.A0006
+      command: "readAttribute"
+      attribute: "SceneTableSize"
+      response:
+          values:
+              - name: "SceneTableSize"
+                saveAs: maxScenes
+          constraints:
+              type: uint16
+
     - label:
-          "preparation step for using commands from Groups cluster: TH generates
-          fabric-unique GroupID, GroupName, random key, EpochKey0 and
-          GroupKeySetID. TH sets GroupKeySecurityPolicy = TrustFirst (1)
-          GroupKeyMulticastPolicy = PerGroupID (0) EpochStartTime0 = 0"
-      verification: |
+          "preparation step for using commands from Groups cluster: Add KeySet"
+      cluster: "Group Key Management"
+      endpoint: 0
+      command: "KeySetWrite"
+      arguments:
+          values:
+              - name: "GroupKeySet"
+                value:
+                    {
+                        GroupKeySetID: 0x01a1,
+                        GroupKeySecurityPolicy: 1,
+                        EpochKey0: "\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf",
+                        EpochStartTime0: 0,
+                        EpochKey1: "\xb0\xb1\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\xbb\xbc\xbd\xbe\xbf",
+                        EpochStartTime1: 0,
+                        EpochKey2: "\xc0\xc1\xc2\xc3\xc4\xc5\xc6\xc7\xc8\xc9\xca\xcb\xcc\xcd\xce\xcf",
+                        EpochStartTime2: 0,
+                    }
 
-      disabled: true
+    - label:
+          "Preparation step for using commands from Groups cluster: Write Group
+          Keys"
+      cluster: "Group Key Management"
+      endpoint: 0
+      command: "writeAttribute"
+      attribute: "GroupKeyMap"
+      arguments:
+          value:
+              [
+                  { FabricIndex: 1, GroupId: 0x0001, GroupKeySetID: 0x01a1 },
+                  { FabricIndex: 1, GroupId: 0x0002, GroupKeySetID: 0x01a1 },
+              ]
 
     - label:
           "TH sends a AddGroup command to DUT with the GroupID field set to G1."
```

Comment:
If it's supposed to be G1, not a fixed 0x0001, doesn't it need to be a test variable?  Followup is fine for this; it comes up in a few places in this test.

Reply:
Added G1 in TC_S_2.3 and opened [issue](https://github.com/CHIP-Specifications/chip-test-plans/issues/2748)